### PR TITLE
Add NWBZarrIO.read_nwb convenience function

### LINKED ISSUE #225: [Feature]: Add `NWBZarrIO.read_nwb` method
### What would you like to see added to HDMF-ZARR?

Add a static method `NWBZarrIO.read_nwb()`  which would accept: 1) local path, 2) S3 path, or 3) an open Zarr store and return the corresponding `NWBFile` object. 

This may also require update to `ZarrIO.can_read` to handle S3 path's properly.

### Is your feature request related to a problem?

This is directly related to https://github.com/NeurodataWithoutBorders/pynwb/issues/1974 with the goal to add a `read_nwb` convenience function. 

### What solution would you like?

See above

### Do you have any interest in helping implement the feature?

Yes.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/hdmf-dev/hdmf-zarr/blob/dev/.github/CODE_OF_CONDUCT.md)
- [X] Have you checked the [Contributing](https://github.com/hdmf-dev/hdmf-zarr/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured this change was not already [requested](https://github.com/hdmf-dev/hdmf-zarr/issues)?

### PR DESCRIPTION
**Pull request by oruebel**
## Motivation

Fix #225

## TODO

- [X] Add `NWBZarrIO.read_nwb` 
- [X] Add unit test for reading from S3 via `read_nwb`
- [X] Add unit test for reading a local file via `read_nwb`
- [ ] Add documentation for `read_nwb` to the tutorials

## Checklist

- [X] Did you update CHANGELOG.md with your changes?
- [X] Have you checked our [Contributing](https://github.com/hdmf-dev/hdmf-zarr/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured the PR clearly describes the problem and the solution?
- [X] Is your contribution compliant with our coding style? This can be checked running `ruff` from the source directory.
- [X] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/hdmf-dev/hdmf-zarr/pulls) for the same change?
- [X] Have you included the relevant issue number using "Fix #XXX" notation where XXX is the issue number? By including "Fix #XXX" you allow GitHub to close issue #XXX when the PR is merged.

### CODE REVIEW

**Comment by rly** [comment]
When you are done with this @oruebel, can you work on https://github.com/hdmf-dev/hdmf-zarr/issues/140 and also maybe copy NWB-related tutorials to PyNWB?

**Comment by oruebel** [comment]
🆗

File: src/hdmf_zarr/nwb.py [hunk 1]
```diff
@@ -63,5 +64,30 @@ def export(self, **kwargs):
             kwargs['container'] = nwbfile
             super().export(**kwargs)
 
+        @staticmethod
+        @docval({'name': 'path',
+                 'type': (str, Path, *SUPPORTED_ZARR_STORES),
+                 'doc': 'the path to the Zarr file or a supported Zarr store'},
+                is_method=False)
```
Comment (h-mayorquin) [inline src/hdmf_zarr/nwb.py:71]: Hi, Oliver, I am curios about what this does? (`is_method`)

A naive reading of mine indicates that this is a method?

Comment (h-mayorquin) [reply src/hdmf_zarr/nwb.py:71]: Maybe related to static method? documentation display? throwing guesses here.

Comment (oruebel) [reply src/hdmf_zarr/nwb.py:71]: `is_method=False` tells docval that there is no `self` or `cls` parameter to allow us to use docval with static functions and functions defined outside of the scope of classes.

File: tests/unit/test_fsspec_streaming.py [hunk 2]
```diff
@@ -8,17 +8,20 @@
...
+    def setUp(self):
         # PLACEHOLDER test file from Allen Institute for Neural Dynamics
         # TODO: store a small test file and use it to speed up testing
-        remote_path = (
+        self.s3_aind_path = (
             "s3://aind-open-data/ecephys_625749_2022-08-03_15-15-06_nwb_2023-05-16_16-34-55/"
             "ecephys_625749_2022-08-03_15-15-06_nwb/"
             "ecephys_625749_2022-08-03_15-15-06_experiment1_recording1.nwb.zarr/"
         )
+        self.https_s3_path = "https://dandiarchive.s3.amazonaws.com/zarr/ccefbc9f-30e7-4a4c-b044-5b59d300040b/"
```
Comment (mavaylon1) [inline tests/unit/test_fsspec_streaming.py]: Use this: "https://dandiarchive.s3.amazonaws.com/zarr/7515c603-9940-4598-aa1b-8bf32dc9b10c/" @oruebel

File: tests/unit/test_fsspec_streaming.py [hunk 3]
```diff
@@ -8,17 +8,20 @@
...
 
 class TestFSSpecStreaming(unittest.TestCase):
-    @unittest.skipIf(not HAVE_FSSPEC, "fsspec not installed")
-    def test_fsspec_streaming(self):
+
+    def setUp(self):
         # PLACEHOLDER test file from Allen Institute for Neural Dynamics
         # TODO: store a small test file and use it to speed up testing
-        remote_path = (
+        self.s3_aind_path = (
```
Comment (mavaylon1) [inline tests/unit/test_fsspec_streaming.py:15]:
```
client = DandiAPIClient(api_url="https://api.dandiarchive.org/api")
dandiset = client.get_dandiset(dandiset_id='000719')
asset = dandiset.get_asset_by_path(path="icephys_DirectoryStore_9_29_24/sub-1214579789_ses-1214621812_icephys_DirectoryStore.nwb.zarr")

s3_url = asset.get_content_url(follow_redirects=1, strip_query=True)
```

**Review by rly — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -12,6 +12,7 @@
 * Added test for opening file with consolidated metadata from DANDI. @mavaylon1 [#206](https://github.com/hdmf-dev/hdmf-zarr/pull/206)
 * Add dimension labels compatible with xarray. @mavaylon1 [#207](https://github.com/hdmf-dev/hdmf-zarr/pull/207)
 * Added link_data --> clear_cache relationship to support repacking zarr nwbfiles: [#215](https://github.com/hdmf-dev/hdmf-zarr/pull/215)
+* Added `NWBZarrIO.read_nwb` convenience method to simplify reading an NWB file. @oruebel [#226](https://github.com/hdmf-dev/hdmf-zarr/pull/226)
```
Comment (h-mayorquin) [inline CHANGELOG.md:15]: @mavaylon1, was this released already? it is on the changelog on the 0.9.0 release but the PR was merged a week ago. Is that a changelog error?

Comment (rly) [reply CHANGELOG.md:15]: This has not been released yet. The changelog is a mistake that we will fix. Thanks.

**Pull request merged**